Tests/Client/RequestRecorderURLProtocol.swift

```diff
@@ -34,16 +36,27 @@ class RequestRecorderURLProtocol: URLProtocol {
         return latestRequest
     }
 
+    /// If set, records only requests with `testSessionHeaderKey` header value set to this value. If `nil`, records all requests.
+    static var currentSessionId: String?
+    
     /// Cleans up existing waiters and recorded requests. We have to explictly reset the state because URLProtocols
     /// work with static variables.
     static func reset() {
+        currentSessionId = nil
         latestRequest = nil
         latestRequestExpectation = nil
     }
 
     override class func canInit(with request: URLRequest) -> Bool {
-        latestRequest = request
-        latestRequestExpectation?.fulfill()
+        guard let sessionId = currentSessionId else {
+            // No sessionId set, record all requests
+            record(request: request)
+            return false
+        }
+        
+        if sessionId == request.value(forHTTPHeaderField: testSessionHeaderKey) {
+            record(request: request)
+        }
```

Comment:
Wouldn't it make sense to make this with `guard` above:
```swift
if let sessionId = currentSessionId, request.value(...) == sessionId {
  record
}
return false
```

Reply:
I think you will need to write me the complete implementation because I don't know what you mean. The logic is:
- if `currentSessionId` is not set, record everything
- if set, record only requests with the same session id

Reply:
Yeah I just realized I got confused. What I suggested was just another way of doing this, not necessarily better:
```swift
if let sessionId = currentSessionId {
  if sessionId = request.value(..) {
    record
  }
} else {
  record
}

return false
```

Let's keep it as it is

Reply:
👍